src/realm/object_id.hpp

```diff
@@ -46,6 +42,9 @@ class ObjectId {
      */
     ObjectId(const char* init) noexcept;
 
+    /**
+     * Constructs an ObjectId with the specified inputs, and a random number
+     */
     ObjectId(Timestamp d, int machine_id = 0, int process_id = 0) noexcept;
```

Comment:
It seems a bit dangerous to allow implicit ObjectId <-> Timestamp conversions. Explicit seems fine, but especially since they are lossy conversions, I don't think you want to allow passing a Timestamp to a function expecting an ObjectId or vice versa, since it is likely a typo or thinko.

I think it would be better to make this (and the conversion operator) `explicit` and either use `is_construtible<To, From>` rather than `is_convertable<From, To>` (yes they are reversed 🙁) in `export2`, or use a custom converter between Timestamp and OID

Reply:
Thanks, I have now disabled implicit conversions and added an explicit conversion function in the value comparisons; I agree it's safer.